daft/dataframe/dataframe.py

```diff
@@ -65,6 +67,15 @@
 ManyColumnsInputType = Union[ColumnInputType, Iterable[ColumnInputType]]
 
 
+def broadcast_metrics(f):
+    @wraps(f)
+    def wrapped(self, *args, **kwargs):
+        self._explain_broadcast()
```

Comment:
I still think this is wrong because most of the functions `f` will add to the plan before execution. E.g. write_parquet will add a write parquet node to the plan.